articles/ros2_dds_security.md

```diff
@@ -0,0 +1,220 @@
+
+If using the directory tree approach to organize security files, RCL supports two different methods for looking up a given node instance's security files in the tree:
+
+- **Exact**: Only load security files from a directory exactly matching the fully-qualified name of the node instance.
+For example, given a node named "baz" within the "/foo/bar/" namespace, only load security files from `<root>/foo/bar/baz/`.
+This is the default behavior.
+- **Prefix**: Attempt to load the most specific set of security files, but if they can't be found, check for security files under a less-specific node name.
+For example, given a node named "baz_123" within the "/foo/bar/" namespace, load security files from `<root>/foo/bar/baz_123/`.
+However, if that directory doesn't exist, find the most specific (i.e. longest) node name that _does_ have security files within that namespace (e.g. `<root>/foo/bar/baz_12/`, or `<root>/foo/bar/baz/`, etc.).
+Note that it will not search higher in the namespace hierarchy.
+
+The desired lookup method can be specified by setting the `$ROS_SECURITY_LOOKUP_TYPE` environment variable to "MATCH_EXACT" (case-sensitive) for the **Exact* method, or "MATCH_PREFIX" (case-sensitive) for the **Prefix** method.
+
+
+### Support for both permissive and strict enforcement of security
+
+Nodes with the security features enabled will not communicate with nodes that don't, but what should RCL do if one tries to launch a node that has no discernable keys/permissions/etc.? It has two options:
+
+- **Permissive mode**: Try to find security files, and if they can't be found, launch the node without enabling any security features.
+This is the default behavior.
+- **Strict mode**: Try to find security files, and if they can't be found, fail to run the node.
+
+The type of mode desired can be specified by setting the `$ROS_SECURITY_STRATEGY` environment variable to "Enforce" (case-sensitive) for strict mode, and anything else for permissive mode.
+
+
+### Support for a master "on/off" switch for all SROS 2 features
+
+In addition to the supported features just discussed, RCL also supports a master shutoff for security features for easy experimentation.
+If it's turned off (the default), none of the security features will be enabled.
+
+In order to enable SROS 2, set the `$ROS_SECURITY_ENABLE` environment variable to "true" (case-sensitive).
+To disable, set to any other value.
+
+
+## Features in the SROS 2 CLI
+
+Configuring a ROS 2 system to be secure in RCL involves a lot of new technology (PKI, DDS governance and permissions files and their syntax, etc.).
+If a user is comfortable with these technologies, the above information should be all that's necessary to properly lock things down.
+However, the [SROS 2 CLI](https://github.com/ros2/sros2) includes a number of commands to help those who don't want to set it all up themselves.
+This section is not meant to be a tutorial, but a reference of capabilities that are available.
```

Comment:
I think it's worth doing, but only once the tutorial is somewhere that [_isn't_ a github repo](https://github.com/ros2/sros2/blob/master/SROS2_Linux.md), otherwise it's way too easy for that link to break.

Reply:
We could make this section a little more abstract; similar to the [CLI section for Actions](http://design.ros2.org/articles/actions.html#introspection-tools). Rather than adding command signatures (that are subject to change), we could list the features that we want from a CLI tool with the descriptions.
Just a suggestion, I don't feel strongly either way.

Reply:
@jacobperron you may have a good point. I reworked this section to similar to the Actions doc, basically a list of requirements to be met. What do you think? Too abstract?

Reply:
LGTM :+1: